hazelcast/src/test/java/com/hazelcast/client/multimap/ClientMultiMapTest.java

```diff
@@ -65,6 +78,137 @@ public void testPut() {
+                .setBinary(false);
+        MultiMapConfig multiMapConfig2 = new MultiMapConfig()
+                .setName("testMultiMapSet")
+                .setValueCollectionType(MultiMapConfig.ValueCollectionType.SET)
+                .setBinary(false);
+        MultiMapConfig multiMapConfig3 = new MultiMapConfig()
+                .setName("expectedMultiMap1")
+                .setValueCollectionType(MultiMapConfig.ValueCollectionType.LIST)
+                .setBinary(false);
+        client.getConfig()
+                .addMultiMapConfig(multiMapConfig1)
+                .addMultiMapConfig(multiMapConfig2)
+                .addMultiMapConfig(multiMapConfig3);
+    }
+
+    public void testMultiMapPutAllTemplate(Map<String, Collection<? extends Integer>> expectedMultiMap,
+                                           Consumer<MultiMap<String, Integer>> putAllOperation1) throws InterruptedException {
+        MultiMap<String, Integer> mmap1 = client.getMultiMap("testMultiMapList");
+        MultiMap<String, Integer> mmap2 = client.getMultiMap("testMultiMapSet");
+        Map<String, Collection<Integer>> resultMap1 = new HashMap<>();
+        Map<String, Collection<Integer>> resultMap2 = new HashMap<>();
+
+        int totalItems = 0;
+        Set<String> ks = expectedMultiMap.keySet();
+        for (String s : ks) {
+            Collection expectedCollection = expectedMultiMap.get(s);
+            totalItems += expectedCollection.size()
+                    + ((Long) expectedCollection.stream().distinct().count()).intValue();
+        }
+
+        final CountDownLatch latchAdded = new CountDownLatch(totalItems);
+        mmap1.addEntryListener(putAllEntryListenerBuilder((event) -> {
+                    String key = (String) event.getKey();
+                    Integer value = (Integer) event.getValue();
+                    Collection<Integer> c;
+                    if (!resultMap1.containsKey(key)) {
+                        c = new ArrayList<>();
+                    } else {
+                        c = resultMap1.get(key);
+                    }
+                    c.add(value);
+                    resultMap1.put(key, c);
+                    latchAdded.countDown();
+                }
+        ), true);
+        mmap2.addEntryListener(putAllEntryListenerBuilder((event) -> {
+                    String key = (String) event.getKey();
+                    Integer value = (Integer) event.getValue();
+                    Collection<Integer> c;
+                    if (!resultMap2.containsKey(key)) {
+                        c = new ArrayList<>();
+                    } else {
+                        c = resultMap2.get(key);
+                    }
+                    c.add(value);
+                    resultMap2.put(key, c);
+                    latchAdded.countDown();
+                }
+        ), true);
+
+        putAllOperation1.accept(mmap1);
+        putAllOperation1.accept(mmap2);
+        latchAdded.await(5, TimeUnit.MINUTES);
```

Comment:
Please use the standard one as follows:
```suggestion
        assertOpenEventually(latchAdded);
```